cpp/src/arrow/util/iterator_test.cc

```diff
@@ -145,6 +145,23 @@ void AssertIteratorNext(T expected, Iterator<T>& it) {
   ASSERT_EQ(expected, actual);
 }
 
+template <typename T>
+Iterator<T> FailsAt(Iterator<T> source, int failure_index, Status failure) {
+  struct Iter {
+    Result<T> Next() {
+      if (index++ == failure_index) {
+        return failure;
+      }
+      return source.Next();
+    }
+    Iterator<T> source;
+    int failure_index;
+    Status failure;
+    int index;
+  };
+  return Iterator<T>(Iter{std::move(source), failure_index, std::move(failure), 0});
+}
+
```

Comment:
If I'm not mistaken, FailsAt and AssertBufferIteratorMatch are now entirely unused?

Reply:
Correct.  I've removed them.